nixos/modules/hardware/video/nvidia.nix

```diff
@@ -284,6 +284,10 @@ in
       source = "${nvidia_x11.bin}/share/nvidia/nvidia-application-profiles-rc";
     };
 
+    # 'nvidia_x11' installs it's files to /run/opengl-driver/...
+    environment.etc."egl/egl_external_platform.d".source =
+      "/run/opengl-driver/share/egl/egl_external_platform.d/";
```

Comment:
```suggestion
    environment.etc."egl/egl_external_platform.d".source = "/run/opengl-driver/share/egl/egl_external_platform.d/";
```

Reply:
Given this is a proprietary binary component, we have to play with the hand we've been dealt.

Reply:
The only alternative that I've identified, as mentioned, is https://github.com/NVIDIA/egl-wayland/issues/39#issuecomment-927288015 which feels like an inferior option for multiple reasons.

Reply:
Settings an ENV to the right path in the nix store sounds like a perfect solution to me which also works without nixos.

Reply:
We're way, way, way beyond any GL stuff working right in not-NixOS. I'm not interested in that even remotely being a goal of this PR. Using an env var over a path carries its own problems, including things like environment propagating differently, rather than symlinks updating instantly, etc.

Reply:
Doesn't seem like the env var way works anyways. I would expect

`
env __EGL_EXTERNAL_PLATFORM_CONFIG_FILENAMES=/run/opengl-driver/share/glvnd/egl_vendor.d/nvidia_wayland.json chromium -enable-features=UseOzonePlatform -ozone-platform=wayland
`

to work without this PR, but it shows as using swiftshader instead of nvidia. Same result with `__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS`.